src/components/pie-chart/chart-legend-content.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+interface ChartLegendContentProps {
+	className?: string;
+	hideIcon?: boolean;
+	payload?: {
+		color: string;
+		value: string | number;
+		[key: string]: string | number;
+	}[];
+	verticalAlign?: 'top' | 'bottom';
+	nameKey?: string;
+}
+
+const ChartLegendContent = React.forwardRef<
+	HTMLDivElement,
+	ChartLegendContentProps
+>(
+	(
+		{
+			className,
+			hideIcon = false,
+			payload = [], // array of objects that represents the data for each legend item
+			verticalAlign = 'bottom', // top | bottom
+			nameKey = 'value',
+		},
+		ref
+	) => {
+		if ( ! payload.length ) {
+			return null;
+		}
+
+		return (
+			<div
+				ref={ ref }
+				className={ `flex items-center justify-center gap-4 ${ verticalAlign === 'top' ? 'pb-3' : 'pt-3' } ${ className }` }
+			>
+				{ payload.map( ( item ) => (
+					<div key={ item.value } className="flex items-center gap-1.5">
+						{ ! hideIcon && (
+							<div
+								className="h-2 w-2 shrink-0 rounded-[2px]"
```

Comment:
Since we are using a `2px` border-radius and a `0.5px` border, it's better to add them to the tailwind config and then use them instead.